src/main/java/it/pagopa/interop/probing/statistics/api/service/impl/StatisticServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package it.pagopa.interop.probing.statistics.api.service.impl;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import it.pagopa.interop.probing.statistics.api.service.StatisticService;
+import it.pagopa.interop.probing.statistics.api.service.TimestreamService;
+import it.pagopa.interop.probing.statistics.api.util.logging.Logger;
+import it.pagopa.interop.probing.statistics.dtos.EserviceStatus;
+import it.pagopa.interop.probing.statistics.dtos.PercentageContent;
+import it.pagopa.interop.probing.statistics.dtos.StatisticContent;
+import it.pagopa.interop.probing.statistics.dtos.StatisticsEserviceResponse;
+
+@Service
+public class StatisticServiceImpl implements StatisticService {
+
+  @Autowired
+  private TimestreamService timestreamService;
+
+  @Autowired
+  private Logger logger;
+
+  @Override
+  public StatisticsEserviceResponse findStatistics(Long eserviceRecordId, Integer pollingFrequency)
+      throws IOException {
+    logger.logRequest(eserviceRecordId, pollingFrequency);
+    List<StatisticContent> content =
+        timestreamService.findStatistics(eserviceRecordId, pollingFrequency);
+    content = cleanStatistics(content);
+    List<PercentageContent> percenteges = calculatePercentages(content);
+    return StatisticsEserviceResponse.builder().values(content).percentages(percenteges).build();
+  }
+
+  private List<StatisticContent> cleanStatistics(List<StatisticContent> content) {
+    List<StatisticContent> falseND = new ArrayList<>();
+    for (StatisticContent statistic : content) {
+      if (statistic.getStatus().equals(EserviceStatus.N_D)) {
```

Comment:
in timestream we do not have ND status, right?

Reply:
No, basically the query fills the gap where we have no OK/KO with the N/D.